app/src/main/java/org/mozilla/fenix/settings/address/AddressManagementFragment.kt

```diff
@@ -0,0 +1,164 @@
+import org.mozilla.fenix.components.StoreProvider
+import org.mozilla.fenix.compose.list.IconListItem
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.address.controller.DefaultAddressManagementController
+import org.mozilla.fenix.settings.address.interactor.AddressManagementInteractor
+import org.mozilla.fenix.settings.address.interactor.DefaultAddressManagementInteractor
+import org.mozilla.fenix.settings.autofill.AutofillAction
+import org.mozilla.fenix.settings.autofill.AutofillFragmentState
+import org.mozilla.fenix.settings.autofill.AutofillFragmentStore
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Displays a list of saved addresses.
+ */
+class AddressManagementFragment : Fragment() {
+
+    private lateinit var store: AutofillFragmentStore
+    private lateinit var interactor: AddressManagementInteractor
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        store = StoreProvider.get(this) {
+            AutofillFragmentStore(AutofillFragmentState(creditCards = emptyList()))
+        }
+
+        interactor = DefaultAddressManagementInteractor(
+            controller = DefaultAddressManagementController(
+                navController = findNavController()
+            )
+        )
+
+        loadAddresses()
+
+        return ComposeView(requireContext()).apply {
+            setContent {
+                FirefoxTheme {
+                    val addresses = store.observeAsComposableState { state -> state.addresses }
+
+                    AddressList(
+                        addresses = addresses.value ?: emptyList(),
+                        onAddressClick = interactor::onSelectAddress,
+                        onAddAddressButtonClick = interactor::onAddAddressButtonClick
+                    )
+                }
+            }
+        }
+    }
+
+    /**
+     * Fetches all the addresses from the autofill storage and updates the
+     * [AutofillFragmentStore] with the list of addresses.
+     */
+    private fun loadAddresses() {
+        lifecycleScope.launch {
+            val addresses = requireContext().components.core.autofillStorage.getAllAddresses()
+
+            lifecycleScope.launch(Dispatchers.Main) {
+                store.dispatch(AutofillAction.UpdateAddresses(addresses))
+            }
+        }
+    }
+}
+
+/**
+ * A list of addresses.
+ *
+ * @param addresses A list of [Address] to display.
+ * @param onAddressClick Invoked when the user clicks on an address.
+ * @param onAddAddressButtonClick Invoked when the user clicks on the "Add address" button.
+ */
+@Composable
+fun AddressList(
+    addresses: List<Address>,
+    onAddressClick: (Address) -> Unit,
+    onAddAddressButtonClick: () -> Unit,
+) {
+    LazyColumn {
+        items(addresses) { address ->
+            TextListItem(
```

Comment:
The description in `TextListItem` uses 14sp instead of 12sp as noted in the Figma https://www.figma.com/file/cSKRUEvfJSzNBv2Mp65Pdv/Mobile-CC-Autofill?node-id=1183%3A34822, but based on https://github.com/mozilla-mobile/fenix/pull/24912#issuecomment-1106831745, we got the okay to bump this up to 14sp so that we can reuse the ListItem component.

<img width="386" alt="Screen Shot 2022-04-22 at 4 39 40 PM" src="https://user-images.githubusercontent.com/1190888/164790589-b45a65ec-4afb-4aec-8255-ad8008960256.png">